hash.go

```diff
@@ -382,3 +384,29 @@ func (db *RoseDB) HIncrBy(key, field []byte, incr int64) (int64, error) {
 	}
 	return valInt64, nil
 }
+
+// GetHashKeys get all stored keys of type Hash.
+func (db *RoseDB) GetHashKeys() (keys [][]byte, err error) {
+	db.hashIndex.mu.RLock()
+	defer db.hashIndex.mu.RUnlock()
+	
+	ts := time.Now().Unix()
+	for key, idxTree := range db.hashIndex.trees {
+
+		rowValue := idxTree.Get([]byte(key))
```

Comment:
The idxTree does not save any info of the key, but it saves the encode key(key+field).
So I think no need to get from idxTree by this key.

The following other data types are the same.

Reply:
I see. here I want to get the expiredAt properties, in fact need to Field to get right info from idxTree.
I will review the code to get right info.

Reply:
fix hash and list bug. set and zset diff logic, should be ok.